v2/pink-sb/src/lib/bottom-sheet/Default.svelte

```diff
@@ -0,0 +1,135 @@
+<script lang="ts">
+    import { Divider } from '$lib/index.js';
+    import type { $$Props } from '$lib/bottom-sheet/index.js';
+
+    export let isOpen: $$Props['isOpen'] = false;
+    export let useSlots: $$Props['useSlots'] = false;
+    export let sheetContainerRef: $$Props['sheetContainerRef'];
+    export let showDivider: $$Props['showDivider'] = true;
+    let isDisappearing: null | boolean = null;
+
+    function setIfDisappearing(isOpenState: boolean) {
+        if (isDisappearing === null) {
+            isDisappearing = false;
+        } else if (!isOpenState) {
+            isDisappearing = true;
+            setTimeout(() => {
+                isDisappearing = false;
+            }, 400);
+        }
+    }
+
+    $: setIfDisappearing(isOpen);
+</script>
+
+{#if isOpen || isDisappearing}
+    <div
+        class="overlay"
+        class:disappear={isDisappearing}
+        on:click={() => {
+            isOpen = false;
+        }}
+        aria-hidden="true"
+    />
+
+    <div class="sheet" bind:this={sheetContainerRef}>
+        {#if useSlots}
+            <div class="content">
+                <slot name="top" />
+            </div>
+            {#if showDivider}<Divider />{/if}
+            <div class="content">
+                <slot name="bottom" />
+            </div>
+        {:else}
+            <div class="content">
+                <slot />
+            </div>
+        {/if}
+    </div>
+{/if}
+
+<style lang="scss">
+    .overlay {
+        position: fixed;
+        width: 100vw;
+        height: 100vh;
+        z-index: 50;
+        top: 0;
+        left: 0;
+        animation: fadeIn 0.4s ease-in-out forwards;
+    }
+
+    :global(.overlay.disappear) {
+        animation: fadeOut 0.4s ease-in-out forwards;
+    }
+
+    @keyframes fadeIn {
+        from {
+            background-color: transparent;
+        }
+        to {
+            background-color: var(--color-overlay-scrim-weak, rgba(25, 25, 28, 40%));
+        }
+    }
+    @keyframes fadeOut {
+        from {
+            background-color: var(--color-overlay-scrim-weak, rgba(25, 25, 28, 40%));
+        }
+        to {
+            background-color: transparent;
+        }
+    }
+
+    @keyframes slideIn {
+        from {
+            bottom: calc(-1 * var(--base-36));
+            opacity: 0;
+        }
+        to {
+            bottom: var(--base-36);
+            opacity: 1;
+        }
+    }
+
+    @keyframes slideOut {
+        from {
+            bottom: var(--base-36);
+            opacity: 1;
+        }
+        to {
+            bottom: calc(-1 * var(--base-36));
+            opacity: 0;
+        }
+    }
```

Comment:
We might be able to simplify some of the CSS by using svelte transitions :thinking: 
https://v4.svelte.dev/docs/svelte-transition